### ISSUE TITLE
Keyboard navigation does not always respect whether new date is "selectable"

### ISSUE THREAD

**Issue by 6eDesign** [6eDesign/svelte-calendar#2]
The existing implementation of keyboard navigation logic is insufficient in certain circumstances, especially when navigating through weeks (up & down arrow keys) and through months (page up and page down).  

In both cases I _think_ we should find the first/last selectable day in the week/month, but I also suspect this might prove awkward/unexpected in certain cases when navigating via up/down through weeks. 

If we want to support more complicated day-by-day 'selectable' logic (ex: weekends aren't selectable or specific days between {start} and {end} are unavailable), the +/-1 (left & right arrow) will also need to follow this pattern.

**Comment by 6eDesign** [comment]
Implemented this by separating 'selected' from 'highlighted' and allowing all visible days to be highlighted but only selectable days to be selected.  I think this is less confusing to a user than skipping over non-selectable days and it's certainly easier to implement.  I may want to add a visual cue to indicate if the highlighted date is non-selectable.
